src/components/Reports/HealthIndicatorReport/HealthIndicatorFormula/HealthIndicatorFormula.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { Box, Card, Skeleton, Typography } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { DateTime } from 'luxon';
+import { useTranslation } from 'react-i18next';
+import { useHealthIndicatorFormulaQuery } from './HealthIndicatorFormula.generated';
+
+const StyledBox = styled(Box)(({ theme }) => ({
+  display: 'flex',
+  gap: 2,
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  marginBottom: theme.spacing(2),
+}));
+
+interface HealthIndicatorFormulaProps {
+  accountListId: string;
+  setNoHealthIndicatorData: Dispatch<SetStateAction<boolean>>;
+}
+
+export const HealthIndicatorFormula: React.FC<HealthIndicatorFormulaProps> = ({
+  accountListId,
+  setNoHealthIndicatorData,
+}) => {
+  const { t } = useTranslation();
+
+  const { data, loading } = useHealthIndicatorFormulaQuery({
+    variables: {
+      accountListId,
+      month: DateTime.now().startOf('month').toISODate(),
+    },
+  });
+
+  const latestMpdHealthData = data?.healthIndicatorData[0];
+
+  if (!data?.healthIndicatorData?.length && !loading) {
+    setNoHealthIndicatorData(true);
```

Comment:
I don't think you're supposed to set state in the render body. Can you put this in a `useEffect`?